src/main/java/dev/jlibra/example/TransferExample.java

```diff
@@ -88,6 +88,8 @@ public static void main(String[] args) throws Exception {
 
         SubmitTransactionResponse response = stub.submitTransaction(submitTransactionRequest);
 
+        Thread.sleep(2000); // add sleep to prevent premature closing of channel
```

Comment:
I saw issue submitted earlier related to this. I'm quite surprised this happens because `submitTransaction` is a blocking call.

Reply:
Yes, what's even more weird is that the response has been received also at this point. Would be interesting to know more about this, but I have not witnessed the issue. Where did you see this problem mentioned?

Reply:
I had this problem before (could reproduce with positive and negative tests), not always but quite often. @Ade92 had this same problem and Thread.sleep solved it for him as well: https://github.com/ketola/jlibra/issues/10

I had similar problems already with gRPC, so that's why I tried this. Usually this is never recognized as applications rarely exit. But I don't know how this can happen, just assume a Threading / not joining Threads before exiting issue.